optuna/visualization/_rank.py

```diff
@@ -0,0 +1,396 @@
+        ]
+
+    return _RankPlotInfo(
+        params=params,
+        sub_plot_infos=sub_plot_infos,
+        target_name=target_name,
+        zs=target_values,
+        color_idxs=color_idxs,
+        has_custom_target=has_custom_target,
+    )
+
+
+def _get_rank_subplot_info(
+    trials: List[FrozenTrial],
+    target_values: np.ndarray,
+    color_idxs: np.ndarray,
+    x_param: str,
+    y_param: str,
+) -> _RankSubplotInfo:
+    xaxis = _get_axis_info(trials, x_param)
+    yaxis = _get_axis_info(trials, y_param)
+
+    filtered_ids = np.array(
+        [
+            i
+            for i in range(len(trials))
+            if x_param in trials[i].params and y_param in trials[i].params
+        ]
+    )
+    filtered_trials = [trials[i] for i in filtered_ids]
+    xs = [trial.params[x_param] for trial in filtered_trials]
+    ys = [trial.params[y_param] for trial in filtered_trials]
+    zs = target_values[filtered_ids]
+    color_idxs = color_idxs[filtered_ids]
+    return _RankSubplotInfo(
+        xaxis=xaxis,
+        yaxis=yaxis,
+        xs=xs,
+        ys=ys,
+        trials=filtered_trials,
+        zs=np.array(zs),
+        color_idxs=np.array(color_idxs),
+    )
+
+
+def _get_axis_info(trials: List[FrozenTrial], param_name: str) -> _AxisInfo:
+    values: List[Union[str, float, None]]
+    is_numerical = _is_numerical(trials, param_name)
+    if is_numerical:
+        values = [t.params.get(param_name) for t in trials]
+    else:
+        values = [
+            str(t.params.get(param_name)) if param_name in t.params else None for t in trials
+        ]
+
+    min_value = min([v for v in values if v is not None])
+    max_value = max([v for v in values if v is not None])
+
+    if _is_log_scale(trials, param_name):
+        min_value = float(min_value)
+        max_value = float(max_value)
+        padding = (math.log10(max_value) - math.log10(min_value)) * PADDING_RATIO
+        min_value = math.pow(10, math.log10(min_value) - padding)
+        max_value = math.pow(10, math.log10(max_value) + padding)
+        is_log = True
+        is_cat = False
+
+    elif is_numerical:
+        min_value = float(min_value)
+        max_value = float(max_value)
+        padding = (max_value - min_value) * PADDING_RATIO
+        min_value = min_value - padding
+        max_value = max_value + padding
+        is_log = False
+        is_cat = False
+
+    else:
+        unique_values = set(values)
+        span = len(unique_values) - 1
+        if None in unique_values:
+            span -= 1
+        padding = span * PADDING_RATIO
+        min_value = -padding
+        max_value = span + padding
+        is_log = False
+        is_cat = True
+
+    return _AxisInfo(
+        name=param_name,
+        range=(min_value, max_value),
+        is_log=is_log,
+        is_cat=is_cat,
+    )
+
+
+def _get_rank_subplot(
+    info: _RankSubplotInfo, target_name: str, print_raw_objectives: bool
+) -> "Scatter":
+    colormap = "RdYlBu_r"
```

Comment:
How about using `RdYlBu` for maximizing?

Reply:
I think using `RdYlBu_r` for both maximizing and minimizing would be more intuitive, because higher values correspond to red and lower values correspond to blue (which meets our intuition as temperature).
In my opinion, it would be somewhat counterintuitive if it is reversed.
Also, we don't know whether the target is maximized or minimized when a custom `target` is specified.

Reply:
OK, let's leave it in this PR and discuss it later.